### ISSUE TITLE
[Crash]: Changing year cause the game to crash and to freeze when using AI

### ISSUE THREAD

**Issue by wensimehrp** [OpenTTD/OpenTTD#12027]
### Version of OpenTTD

14.0 beta 3, Windows 11

### Steps to reproduce

1. Start a new game using an AI
2. Change the year via sandbox options
3. Crash

### Upload crash files

[crash20240206.zip](https://github.com/OpenTTD/OpenTTD/files/14188562/crash20240206.zip)

**Comment by glx22** [comment]
```
 	[Cadre en ligne] openttd.exe!google_breakpad::ExceptionHandler::WriteMinidump() Ligne 747	C++
 	openttd.exe!google_breakpad::ExceptionHandler::WriteMinidump(const std::wstring & dump_path, bool(*)(const wchar_t *, const wchar_t *, void *, _EXCEPTION_POINTERS *, MDRawAssertionInfo *, bool) callback, void * callback_context, _MINIDUMP_TYPE dump_type) Ligne 773	C++
 	openttd.exe!CrashLogWindows::WriteCrashDump() Ligne 113	C++
 	[Cadre en ligne] openttd.exe!std::_Func_class<bool>::operator()() Ligne 883	C++
 	openttd.exe!CrashLogWindows::TryExecute(std::basic_string_view<char,std::char_traits<char>> section_name, std::function<bool __cdecl(void)> && func) Ligne 124	C++
 	openttd.exe!CrashLog::MakeCrashLog() Ligne 290	C++
 	openttd.exe!ExceptionHandler(_EXCEPTION_POINTERS * ep) Ligne 341	C++
 	[Code externe]	
 	openttd.exe!raise(int signum) Ligne 547	C++
 	openttd.exe!abort() Ligne 71	C++
 	openttd.exe!FatalErrorI(const std::string & str) Ligne 143	C++
 	openttd.exe!AssertFailedError(int line, const char * file, const char * expression) Ligne 20	C++
>	openttd.exe!DeleteStaleLinks(Station * from) Ligne 3895	C++
 	openttd.exe!OnTick_Station() Ligne 4004	C++
 	openttd.exe!CallLandscapeTick() Ligne 1681	C++
 	openttd.exe!StateGameLoop() Ligne 1488	C++
 	openttd.exe!GameLoop() Ligne 1610	C++
 	openttd.exe!VideoDriver::GameLoop() Ligne 39	C++
 	openttd.exe!VideoDriver::GameThread() Ligne 48	C++
 	openttd.exe!StartNewThread::__l3::<lambda_1>::operator()(const char * name, void(*)(VideoDriver *) && F, VideoDriver * && <A_0>) Ligne 64	C++
 	[Code externe]	
 	openttd.exe!thread_start<unsigned int (__cdecl*)(void *),1>(void * const parameter) Ligne 97	C++
 	[Code externe]	
```
It's cargodist stuff.

**Comment by 2TallTyler** [comment]
Fixed by #12146.
